### ISSUE TITLE
Compliance/Compliance Obligation - Internal User can view a compliance obligation details

### ISSUE THREAD

**Issue by NicoleGovvy** [bcgov/cas-compliance#71]
#### Description:

As an internal user, I have the ability to view the details of all compliance obligations from the compliance summaries grid.

- Design | [Review 2024 Compliance Obligation Report page](https://www.figma.com/proto/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?page-id=14307%3A1663&node-id=43952-144&p=f&viewport=392%2C-559%2C0.02&t=R1kXjEdKzzMu0SkZ-1&scaling=scale-down&content-scaling=fixed&starting-point-node-id=21296%3A1271&show-proto-sidebar=1) for internal CAS users

Acceptance Criteria:

- [ ] Given that I am on my compliance obligation grid, when I click on `View Details`, I am brought to the compliance obligation report page and I see tabs on the left: Review [year] compliance obligation report & Review penalty summary (when there is)

#### Development Checklist:

 - [ ] Organize the folder structure by adding the `manage-obligation-review-summary` page to the `idir` folder
 - [ ] In the `components/compliance-summary/manage-obligation` folder, create an `internal` folder
 - [ ] Inside `internal`, create a `review-compliance-summary` subfolder
 - [ ] In `manage-obligation/internal/review-compliance-summary`, create the page and component files
 - [ ] Create the RJSF schema for the Review Summary page
 - [ ] Create a new task list for internal users that includes the `manage-obligation-review-summary` and `review-penalty-summary` pages
 - [ ] Reuse the existing getComplianceSummary function to fetch data
 - [ ] Write tests

#### Definition of Ready (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

#### Definition of Done (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

#### Blocked By/Blocking
Blocking #280 

#### Notes:

-

**Comment by vesselak** [comment]
@nanyangpro Would you please update the ticket with the correct wireframe link? I looked though design and can't find the page so we might be missing this one

**Comment by vesselak** [comment]
@fviduya can you please read the ticket and confirm if this work is complete? the figma link is incorrect. it is the internal user compliance summary grid navigation

**Comment by fviduya** [comment]
I am a bit confused about the description and the acceptance criteria not matching. I think this should be the [wireframe for the grid](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=21296-1060&t=U4V2LfW946MvSbFH-0) if we are to follow the AC. 

For the AC:
- [ ] Given that I am on the compliance ~obligation~ summaries grid, when I click on `View Details`, I am brought to the compliance obligation report page, and I see tabs on the left: Review [year] compliance obligation report 
- [x] Given that I am on my compliance obligation grid, when I click on review credits issuance request I see review by analyst, review by director, and track status of issuance - **This is done as well, additional work needs to be done in terms of changing the link name to `Review Credits Issuance Request` #217 **

I'll update the description 🙂

**Comment by vesselak** [comment]
@nanyangpro we can locate the wireframe to the internal view of an obligation. Would you please add it to the ticket?

**Comment by nanyangpro** [comment]
Hi @vesselak (cc: @fviduya) - So sorry I didn't realize this ticket was asking for the **internal** view of an obligation!

We didn't create such screen specifically given the vision that the internal side would simply be the same as what's on the external side, except that:
- all the form field would be display in read-only mode instead of editing mode; and
- all the TaskList sections would have been completed with the green checkmarks (understanding that such feature may have not been implemented yet)

That being said, we do have an [example](https://www.figma.com/proto/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?page-id=14307%3A1663&node-id=35523-6877&p=f&viewport=392%2C-559%2C0.02&t=CY5nL6q6X7STTETm-1&scaling=scale-down&content-scaling=fixed&starting-point-node-id=21296%3A1271&show-proto-sidebar=1) designed to showcase the specific scenario when there's no obligation or earned credits on the internal side. This would give a general idea how we wanted things to be displayed.

**Comment by nanyangpro** [comment]
Hi @vesselak (cc: @fviduya) - As discussed during Jul 15 Dev/Design Sync, this ticket is missing wireframe.

- Changing it into a design ticket as of now.
- Once the design is drafted, refined and approved, will change it back to a dev ticket for refinement and implementation.

**Comment by nanyangpro** [comment]
Updates

- Draft design | [Review 2024 Compliance Obligation Report page](https://www.figma.com/proto/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?page-id=14307%3A1663&node-id=43952-144&p=f&viewport=392%2C-559%2C0.02&t=R1kXjEdKzzMu0SkZ-1&scaling=scale-down&content-scaling=fixed&starting-point-node-id=21296%3A1271&show-proto-sidebar=1) for internal CAS users
- Draft design | [Review Penalty Summary page](https://www.figma.com/proto/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?page-id=14307%3A1663&node-id=44166-151&p=f&viewport=392%2C-559%2C0.02&t=R1kXjEdKzzMu0SkZ-1&scaling=scale-down&content-scaling=fixed&starting-point-node-id=21296%3A1271&show-proto-sidebar=1) for internal CAS users

cc: @vesselak for getting approval from the business area when she's back. (The draft designs have been shown to the developers for feedback already.)

**Comment by vesselak** [comment]
@nanyangpro reviewed designs with Kirk this afternoon and both pages are approved after removing the Generate Invoice button.  This functionality is not required.

**Comment by nanyangpro** [comment]
Perfect, thanks @vesselak. The designs are updated as requested above. The ticket description is updated accordingly, changing it back to a dev ticket for further refinement and implementation.

**Comment by fviduya** [comment]
I have tested this in the DEV environment alongside #91. [Here is the test document for reference](https://buttoninc-my.sharepoint.com/:w:/g/personal/francesca_button_is/EetXSoaaDkxMsa6Ze5IlQV8BThuS0hiLlon4-O0D1lhqmA?e=i66Ba1&nav=eyJoIjoiMjEyNjM5MDI3MiJ9).

I can verify that:
- [x] As an internal user, given that I am on the compliance summaries grid, when the record has an excess emission, I see a `View Details` action
   - [x] When I click on `View Details`, I am brought to the compliance obligation report page and I see tabs on the left: Review [year] Compliance Obligation Report & Review Penalty Summary (when there is)
- [x] As an external user, given that I am on the compliance summaries grid, when the record has an excess emission, I see a `Manage Obligation` action
   - [x] When I click on `Manage Obligation`, I am brought to the compliance obligation report page and I see tabs on the left: Review [year] Compliance Summary, Download Payment Instructions, & Pay Obligation and Track Payment(s)

Moving to Pending PO for review of test. Thank you!

**Comment by vesselak** [comment]
Reviewed testing documentation and accepting ticket
